draft-ietf-quic-transport.md

```diff
@@ -1266,6 +1266,53 @@ actual exponent is one-less than the explicit exponent, and the value represents
 4096 microseconds.  Any values larger than the representable range are clamped
 to 0xFFFF.
 
+
+### ACK Frames and Packet Protection
+
+ACK frames that acknowledge protected packets MUST be carried in a packet that
+has an equivalent level of packet protection.
+
+Packets that are protected with 1-RTT keys MUST be acknowledged in packets that
+are also protected with 1-RTT keys.
+
+A packet that is not protected and claims to acknowledge a packet number that
+was sent with packet protection is not valid.  An unprotected packet that
+carries acknowledgments for protected packets MUST be discarded in its entirety.
+
+Packets that a client sends with 0-RTT packet protection MUST be acknowledged by
+the server in packets protected by 1-RTT keys.  This can mean that the client is
+unable to use these acknowledgments if the server cryptographic handshake
+messages are delayed or lost.  Note that the same limitation applies to other
+data sent by the server protected by the 1-RTT keys.
+
+Unprotected packets, such as those that carry the initial cryptographic
+handshake messages, MAY be acknowledged in unprotected packets.  Unprotected
+packets are vulnerable to falsification or modification.  Unprotected packets
+can be acknowledged along with protected packets in cases where the peer has
+packet protection keys.
+
+An endpoint SHOULD acknowledge packets containing cryptographic handshake
+messages in the next unprotected packet that is sends, unless it is able to
```

Comment:
If the concern is timeliness, why not say immediately, rather than discuss whether it's protected or unprotected?  Also, the vulnerability comes from the peer processing the unprotected ack, so as long as the peer is expected to process it, the sender of the ack doesn't gain much by waiting for 1RTT keys to ack it.

ie: "An endpoint SHOULD acknowledge packets containing cryptographic handshake messages with no ack delay, even if the 1RTT keys are not yet available." 

I wanted to put immediately, but I think it's reasonable for the client to process a 2 or 3 packet reject and then only send 1 ack, if all the packets arrive essentially at once.